src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -153,77 +169,156 @@ export default class SimidMapCreative extends BaseSimidCreative {
-    location: coordinates,
-    name: searchParameter,
-    openNow: true,
-    rankBy: google.maps.places.RankBy.DISTANCE
-  };
-  const service = new google.maps.places.PlacesService(this.map_);
-  service.nearbySearch(request, this.displayResults_.bind(this));
-}
+  /**
+   * Searches for the closest corresponding businesses based off of the given search parameter,
+   * and places pins on the map that represent the 4 closest locations.
+   * @param {String} searchParameter A string with the business's name to use in the query.
+   * @param {!google.maps.LatLng} coordinates The LatLng object of user's current location.
+   * @private 
+   */
+  findNearby_(searchParameter, coordinates) {
+    const request = {
+      location: coordinates,
+      name: searchParameter,
+      openNow: true,
+      rankBy: google.maps.places.RankBy.DISTANCE
+    };
+    const service = new google.maps.places.PlacesService(this.map_);
+    service.nearbySearch(request, this.displayResults_.bind(this));
+    this.createTravelChoices_();
+  }
 
-/**
- * Displays the closest business locations to a user's current location.
- * @param {!Object} results An array of Place Results from the search query.
- * @param {!google.maps.places.PlacesServiceStatus} status The status returned 
- *  by the PlacesService on the completion of its searches.
- * @private 
- */
-displayResults_(results, status) {
+  /**
+   * Displays the closest business locations to a user's current location.
+   * @param {!Object} results An array of Place Results from the search query.
+   * @param {!google.maps.places.PlacesServiceStatus} status The status returned 
+   *  by the PlacesService on the completion of its searches.
+   * @private 
+   */
+  displayResults_(results, status) {
     if (status == google.maps.places.PlacesServiceStatus.OK) {
       for (let i = 0; i < DEFAULT_LOCATION_NUM_DISPLAYED; i++) {
         this.placeMapMarker_(results[i]);
       }
+      this.closestLocation_ = results[0].geometry.location;
+      this.displayDirections_(this.closestLocation_, this.currentLocation_);
     }
-}
+  }
 
-/**
- * Creates and displays a marker on the map representing a given place.
- * @param {!Object} place A Place Result object.
- * @private 
- */
-placeMapMarker_(place) {
-  const placeIcon = {
-    url: this.markerImage_,
-    scaledSize: new google.maps.Size(MARKER_SIZE, MARKER_SIZE)
-  };
-  const placeMarker = new google.maps.Marker({
-    map: this.map_,
-    position: place.geometry.location,
-    icon: placeIcon
-  });
-  google.maps.event.addListener(placeMarker, 'click', () => {
-    const infowindow = new google.maps.InfoWindow;
-    infowindow.setContent(place.name);
-    infowindow.open(this.map_, this);
-  });
-}
+  /**
+   * Creates and displays a marker on the map representing a given place.
+   * @param {!Object} place A Place Result object.
+   * @private 
+   */
+  placeMapMarker_(place) {
+    const placeIcon = {
+      url: this.markerImage_,
```

Comment:
In the beginning you set markerImage_ = null but I don't see you change that anywhere in the code. Where do you take the marker parameter (in the adParams) and set that as this url or do you want me to do that?

Reply:
I set markerImage_ to the marker param from AdParams on [line 86](https://github.com/googleinterns/step33-2020/pull/39/files/29acb97776d77bf670ae1318bd5ea3f8a1975981#diff-6f605ee0c53a273329bab0887a937b2bR86), I changed these variable names in my last PR so that's probably why it didn't show up in this one!

Reply:
ah I see! Thanks for pointing me to that change